### ISSUE TITLE
Duplicated VIPs keep existing on two keepalived nodes

### ISSUE THREAD

**Issue by ydcool** [acassen/keepalived#1702]
**Describe the bug**

We have three nodes in a k8s cluster running keepalived pods, and two of them keep holding one of managed VIPs (10.16.4.60) at the same time, with no state switching logs, no errors, and keep holding even after killing and recreating all keepalived pods.

**To Reproduce**
Any steps necessary to reproduce the behavior:

Not clear yet

**Expected behavior**

Only one node holds the VIP at the same time

**Keepalived version**
Output of `keepalived -v`
```
Keepalived v1.2.24 (02/14,2019)

Copyright(C) 2001-2019 Alexandre Cassen, <acassen@gmail.com>

Build options:  PIPE2 LIBNL3 RTA_ENCAP RTA_NEWDST RTA_PREF RTA_VIA FRA_OIFNAME FRA_SUPPRESS_PREFIXLEN FRA_SUPPRESS_IFGROUP FRA_TUN_ID RTAX_CC_ALGO RTAX_QUICKACK IPV4_DEVCONF LIBIPTC LIBIPSET LVS LIBIPVS_NETLINK IPVS_SYNCD_ATTRIBUTES IPVS_64BIT_STATS VRRP VRRP_AUTH VRRP_VMAC SOCK_NONBLOCK SOCK_CLOEXEC FIB_ROUTING INET6_ADDR_GEN_MODE SNMP_V3_FOR_V2 SNMP SNMP_KEEPALIVED SNMP_CHECKER SNMP_RFC SNMP_RFCV2 SNMP_RFCV3 SO_MARK
```

**Distro (please complete the following information):**
 - Name [Ubuntu]
 - Version [16.04]
 - Architecture [x86_64]

**Details of any containerisation or hosted service (e.g. AWS)**
If keepalived is being run in a container or on a hosted service, provide full details

**Configuration file:**
A full copy of the configuration file, obfuscated if necessary to protect passwords and IP addresses
```
! Configuration File for keepalived

global_defs {
   router_id LVS_DEVEL
   vrrp_skip_check_adv_addr
   vrrp_garp_interval 0
   vrrp_gna_interval 0
}

vrrp_script chk_apiserver {
    script "/etc/keepalived/chk_apiserver.sh"
    interval 30
    fall 2
    rise 1
}

vrrp_instance VI_apiserver_vip {
    state BACKUP
    nopreempt
    garp_master_delay 5
    garp_master_repeat 3
    garp_master_refresh 1
    dont_track_primary
    interface ens4.3012
    virtual_router_id 60
    priority 100
    
    advert_int 1
    
    authentication {
        auth_type AH
        auth_pass KjpyNnhrtQo8ls91
    }
    virtual_ipaddress {
      
10.16.12.60
    }
    track_script {
      chk_apiserver
    }
}

vrrp_instance VI_apiserver_vip_manage {
    state BACKUP
    nopreempt
    garp_master_delay 5
    garp_master_repeat 3
    garp_master_refresh 1
    dont_track_primary
    interface ens3
    virtual_router_id 61
    priority 100
    
    advert_int 1
    
    virtual_ipaddress {
      
10.16.4.60
    }
    track_script {
      chk_apiserver
    }
}
```

**Notify and track scripts**
If any notify or track scripts are in use, please provide copies of them

**System Log entries**
Full keepalived system log entries from when keepalived started
stor-mgt01 holding VIP 10.16.4.60
![image](https://user-images.githubusercontent.com/5668295/88274106-c9ac4280-cd0d-11ea-8763-5ab3a003374a.png)
tcpdump on stor-mgt01
![image](https://user-images.githubusercontent.com/5668295/88274595-89998f80-cd0e-11ea-8fd7-fa206edb19f1.png)

stor-mgt02 holding VIP 10.16.4.60 too
![image](https://user-images.githubusercontent.com/5668295/88274401-39bac880-cd0e-11ea-85c7-d1296afecb04.png)

stor-mgt03 is in normal BACKUP state
![image](https://user-images.githubusercontent.com/5668295/88274220-ec3e5b80-cd0d-11ea-9c82-24739fecd97d.png)

**Did keepalived coredump?**
If so, can you please provide a stacktrace from the coredump, using gdb.

**Additional context**

Is there any clear trouble shooting approaches for this kind of problem?

**Comment by pqarmitage** [comment]
The troubleshooting approach to adopt here is to check your configuration files :)

The configuration file you have shown above looks like the configuration file from stor-mgt01, or possibly stor-mgt03. For vrrp_instance `VI_apiserver_vip_manage` you have configured `router_id 61`. On stor-mgt02, which it looks to me has IP address 10.16.4.46, you have configured the vrrp_instance to have `router_id 91`. These are therefore different virtual routers, and each one will have it's own master, which is what you are experiencing. Note: I cannot determine from the information provided what `router_id` is configured on stor-mgt03.

It appears that you are using the same configurations on each system (with the exception of the incorrect router_id). You should not do this, and each virtual router (i.e. the matching vrrp_instances on the different systems) should have different priorities, otherwise you will get unwanted behaviour if systems start up simultaneously. You have configured `nopreempt` on each vrrp_instance, and therefore no instance will take over as master, even if another instance is currently master with a lower priority.

Quite some time ago I implemented conditional configuration in keepalived, so that you could have, for example:
```
@stor-mgt01  priority 150
@stor-mgt02  priority 200
@stor-mgt03  priority 250
```
and then use the same configuration file on all three systems.

I think, however, that this feature was added some time after v1.2.24, which is very very old. There have been hundreds of enhancements, new features, performance improvements and bug fixes since v1.2.24 and you are strongly advised to upgrade to v2.1.5. It appears that your distro doesn't keep up-to-date with keepalived versions, and so you will need to build keepalived yourself, but that isn't difficult.

**Comment by ydcool** [comment]
@pqarmitage Thank you , we'll upgrade to 2.x and update the configuration file.
